GCTC-NTGC/gc-digital-talent
# Bug - Improve unique email validation

## Summary

This resolves an issue with not being able to update an existing user due to the unique email validation by ignoring the user you are updating.

## Testing

1. Regenerate the queries `npm run codegen --workspaces`
2. Build all projects `npm run production --workspaces`
3. Navigate tot `/admin/users/`
4. Edit a user and update field other than email
5. Ensure you can save the user
6. Edit it again, using an existing email
7. Ensure you get the unique email error
8. Navigate to `/talent/profile`
9. Go to to the "About Me" form and repeat steps 4-7
10. Logout and register as a new user
11. Attempt to create account with an existing email
12. Ensure you get the unique email error
